test/lib/test_backend/test_tchannel_backend.go

```diff
@@ -47,8 +47,14 @@ func BuildTChannelBackends(
 	n := len(knownTChannelBackends)
 	result := make(map[string]*TestTChannelBackend, n)
 
+	var clientName, serviceName string
 	for i := 0; i < n; i++ {
-		serviceName := knownTChannelBackends[i]
+		clientName = knownTChannelBackends[i]
+
+		// TODO: (lu) tmp, get backend thrift service name from config
+		if clientName == "baz" {
```

Comment:
dont know how we are going to get this information.

This problem might go away if we have the ability to register individual endpoints in the backend.